ykrt/src/deopt.rs

```diff
@@ -268,18 +277,82 @@ unsafe extern "C" fn __ykrt_deopt(
     guardid: usize,
     // Return address of deoptimize call. Used to find the correct stackmap record.
     retaddr: usize,
+    // The parent trace's call stack at the time of the guard failure. Required to assemble
+    // a side trace.
+    jitcallstack: *const c_void,
     // Current stack pointer. Needed to read spilled register values from the stack.
     rsp: *const c_void,
 ) -> NewFramesInfo {
     // Put the CompiledTrace back into an Arc, so it is dropped properly.
     let ctr = Arc::from_raw(ctr);
 
+    // Check if we have a side trace and execute it.
+    if guardid != SIDETRACE_LAST_GUARD_ID {
+        let guard = &ctr.guards[guardid];
+        if let Some(st) = guard.getct() {
+            let registers = Registers::from_ptr(rsp);
+            let live_vars = ctr.smap.get(&retaddr.try_into().unwrap()).unwrap();
+            let mut ykctrlpvars = Vec::new();
+            for (_i, locs) in live_vars.iter().skip(1).enumerate() {
+                assert!(locs.len() == 1);
+                let l = locs.get(0).unwrap();
```

Comment:
`let l = locs[0];` should work?

Reply:
`locs` is of type `LiveVar` which is a wrapper around a vec, so unfortunately not.

Reply:
?

[Although my suggestion was wrong. I think `let l = &locs[0];` should work.]

Reply:
`locs` is of type `LiveVar`, which is just a struct. You can't index into a struct.

Reply:
Oh! So then we should (as always) document why the `unwrap` can't fail.

Reply:
https://github.com/ptersilie/yk/blob/e4b4d66bd59709c5955797a78787143d2d2f389d/yksmp/src/lib.rs#L38

Reply:
Ah!